task.h

```diff
@@ -68,4 +86,45 @@ void print_task(task_spec *spec, UT_string *output);
 /* Parse task as printed by print_task. */
 task_spec *parse_task(char *task_string, int64_t task_length);
 
+/*
+ * SCHEDULED TASK: Contains information about a scheduled task:
+ * the task iid, the task specification and the task status
+ * (WAITING, SCHEDULED, RUNNING, DONE) and which node the
+ * task is scheduled on.
+ *
+ */
+
+/* The scheduling_state can be used as a flag when we are listening for an event,
+ * for example TASK_WAITING | TASK_SCHEDULED. */
+enum scheduling_state {
+  TASK_WAITING = 0,
```

Comment:
I think that if we're going to use bitmasks to filter these, we will probably need to start the enum at 1, not 0 (otherwise, we can't differentiate between an event that we didn't register for and TASK_WAITING).